### PR TITLE
Fixed a bug with cleaning up state after a network is disbanded

### PR DESCRIPTION
**Pull request by theJenix**
In an attempt to reproduce #245, I noticed that the guest was not being cleaned up properly.  This should fix that.

### CODE REVIEW

**Comment by theJenix** [comment]
This is forked off of cancel_partial.  We should probably pull that in first.

**Comment by reidbaker** [comment]
Will wait till cancel_partial is pulled in to review

File: SoundStream/project.properties [hunk 1]
```diff
@@ -12,5 +12,5 @@
 
 # Project target.
 target=android-17
-android.library.reference.1=../third_party/SlidingMenuLibrary
-android.library.reference.2=../../../Programs/adt-bundle-linux-x86_64-20130717/sdk/extras/android/support/v7/appcompat
+android.library.reference.1=../../../../../usr/local/var/lib/android-sdk/extras/android/support/v7/appcompat
```
Comment (dgreenhalgh) [inline SoundStream/project.properties]: Is this *nix specific?

File: SoundStream/src/com/thelastcrusade/soundstream/net/ConnectionReader.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class ConnectionReader {
+
+    private InputStream inStream;
+    private Messenger messenger;
+    private String remoteAddress;
+
+    /**
+     * @param remoteDevice 
+     * @param mmMessenger
+     * @param mmInStream
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/ConnectionReader.java]: These variable names don't exactly match the params of this method   ¯_(ツ)_/¯

File: SoundStream/src/com/thelastcrusade/soundstream/net/ConnectionWriteThread.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
+/*
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/ConnectionWriteThread.java:1]: The version of this file from #330 had more descriptive commenting that didn't get merged into this pull request

File: SoundStream/src/com/thelastcrusade/soundstream/net/MessageEnqueuer.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ */
+public class MessageEnqueuer {
+
+    private int mmOutMessageNumber = 1;
+    private ConnectionWriter writer;
+
+    /**
+     * 
+     */
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/MessageEnqueuer.java]: Empty javadoc

File: SoundStream/src/com/thelastcrusade/soundstream/net/MessageThreadMessageDispatch.java [hunk 5]
```diff
@@ -85,19 +85,23 @@ public void handleMessage(Message msg) {
...
         
-        if (msg.what == MessageThread.MESSAGE_READ) {
+        if (msg.what == ConnectionConstants.MESSAGE_READ) {
             messageNo = msg.arg1;
             message   = (IMessage)msg.obj;
-            fromAddr  = msg.getData().getString(MessageThread.EXTRA_ADDRESS);
+            fromAddr  = msg.getData().getString(ConnectionConstants.EXTRA_ADDRESS);
             handleMessage(messageNo, message, fromAddr);
+        } else if (msg.what == ConnectionConstants.MESSAGE_FINISHED) {
+            //TODO: NYI
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/MessageThreadMessageDispatch.java]: When do you plan to implement this?

File: SoundStream/src/com/thelastcrusade/soundstream/net/bt/BluetoothConnection.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+ * This thread is responsible for sending and receiving messages once the connection has been established.
+ * 
+ * 
+ * @author Jesse Rosalia
+ *
+ */
+public abstract class BluetoothConnection extends Thread {
+    private final String TAG = BluetoothConnection.class.getSimpleName();
+
+    private final BluetoothSocket mmSocket;
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/bt/BluetoothConnection.java]: mmmmm :hamburger:

File: SoundStream/src/com/thelastcrusade/soundstream/net/bt/BluetoothConnection.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        try {
+            mmSocket.close();
+        } catch (IOException e) {
+            
+        } finally {
+            //before we exit, notify the launcher thread that the connection is dead
+            onDisconnected();
+        }
+    }
+}
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/net/bt/BluetoothConnection.java]: newline

File: SoundStream/src/com/thelastcrusade/soundstream/service/TransferService.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return new TransferServiceBinder();
+    }
+
+
+    /**
+     * 
+     */
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/service/TransferService.java:95]: Empty javadoc

File: SoundStream/src/com/thelastcrusade/soundstream/service/TransferService.java [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+//                    String fromAddr = intent.getStringExtra(MessagingService.EXTRA_ADDRESS);
+//                    long   songId   = intent.getLongExtra(  MessagingService.EXTRA_SONG_ID, SongMetadata.UNKNOWN_SONG);
+//
+//                    if (songId == SongMetadata.UNKNOWN_SONG) {
+//                        Log.wtf(TAG, "REQUEST_SONG_MESSAGE action received without a valid song id");    
+//                    } else {
+//                        sendSongData(fromAddr, songId);
+//                    }
+//                }
+//            })
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/service/TransferService.java]: I believe that this block was removed in #330

**Comment by dgreenhalgh** [comment]
I reviewed this pull request using #330 as a reference because of how much overlap there is between the code. A couple of the issues seem to have been fixed in #330 but just haven't been merged in to this yet.

**Comment by theJenix** [comment]
Just merged in master (which includes #330).  This should be much easier to review now.

**Comment by dgreenhalgh** [comment]
:+1:

**Pull request merged**